Backend/ServerManagement.Tests/Azure/AzureServerManagerTests.cs

```diff
@@ -0,0 +1,113 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Moq;
+using ServerManagement.Azure;
+using ServerManagement.Azure.Configuration;
+using ServerManagement.Core.Exceptions;
+using ServerManagement.Core.Models;
+
+namespace ServerManagement.Tests.Azure;
+
+public class AzureServerManagerTests
+{
+    private readonly Mock<ILogger<AzureServerManager>> _mockLogger;
+    private readonly Mock<IOptions<AzureVmConfiguration>> _mockVmOptions;
+    private readonly Mock<IOptions<GameServerConfiguration>> _mockGameOptions;
+    private readonly Mock<HttpClient> _mockHttpClient;
+    private readonly AzureVmConfiguration _vmConfig;
+    private readonly GameServerConfiguration _gameConfig;
+
+    public AzureServerManagerTests()
```

Comment:
This file needs more documentation. Document it for the perspective of someone that is familiar with the concept of unit tests, but not specifically familiar with mock